### ISSUE TITLE
Xiaomi Aqara square wireless buttons stopped working

### ISSUE THREAD

**Issue by llonca13** [Danielhiversen/PyXiaomiGateway#161]
Using latest stable 0.99.3 but this is broken for the last two months, and I always update as soon as a new version comes out. Sorry can be more specific.
**Home Assistant release with the issue (if applicable):**

Aqara Square button (1st and 2nd gen) automations in HA stopped working. I see in history that devices become Unavailable and then recognized again in approx every 12hrs or so. But even when I can see their state as 'off' (they are allways off, they work as binary_sensor component), they still don't trigger automations. Logs don't say anything, as if nothing happens. I tried pairing - repairing - battery changing - everything. I use Xiaomi Hub on latest FW. I also have many Aqara Wireless switches that work perfectly with the same component and syntax.

Alot of people providing the same problems on reddit:
https://www.reddit.com/r/homeassistant/comments/curh39/aqara_button_square_with_gyro_stopped_working/
https://www.reddit.com/r/homeassistant/comments/dct42j/xiaomi_binary_sensor_wireless_button_automations/
**Last working Home Assistant release (if known):**

Any more information I can provide I will. Thank you!

**PyXiaomiGateway library version (if known):**

**Operating environment (Hass.io/Docker/Windows/etc.):**

**Traceback (if applicable):**
```

```

**Description of the bug:**

**Comment by syssi** [comment]
It could be a docker issue because my setups (3x RPi3 + venv) doesn't have this issue. How many devices do you use / have you paired?

**Comment by llonca13** [comment]
4x non-wireless aqara swicthes
4x wireless aqara switches
1x round xiaomi wireless button 
3x aqara square wireless buttons (not-working anymore)
2x motion sensors
3x door sensors
6x temperature sensors
total : 23

Also, I have been using them for almos a year before they stopped working, without any issue. Since I've been very busy at work for a long time now I haven't make any change(no new devices, no config changed) in my HA for at least 6 months except updating HA version and/or fixing breaking change syntax in some platforms.
Thanks!

**Comment by llonca13** [comment]
I could spin up my dusty rpi3 with hass.io and my configs to see if that make any difference? Or is hass.io HA also docker based?

**Comment by syssi** [comment]
Hass.io is docker based, too. You could follow this guide on any linux host (python >=3.6):

https://www.home-assistant.io/docs/installation/virtualenv/

**Comment by llonca13** [comment]
Will try to do it later today, and will post back results.

**Comment by llonca13** [comment]
I've tried virtualenv installation using my config files from unraid, on my rpi4 (I've stopped my unradi docker before that of course). Results are the same, everything works except my aqara square buttons.

I checked now, and all my buttons are that aren't working are 2nd gen WXKG12LM version. Could you please check do you have the same version? Maybe this bug is specific to this hw revision.

**Comment by llonca13** [comment]
This time all my aqara buttons were reporting to HA normally (state was "off", not "unavailable") , but when I press them no automations triggers and I can't see nothing in the logs, same like before.

**Comment by syssi** [comment]
Please enable the debug output of this library and the Home Assistant component by:

```
# configuration.yaml

logger:
  default: warn
  logs:
    xiaomi_gateway: debug
    homeassistant.components.xiaomi_aqara: debug
    homeassistant.components.binary_sensor.xiaomi_aqara: debug
```

And try to reproduce the issue: Restart HA, toggle the button and post the relevant parts of your home-assistant.log.

**Comment by llonca13** [comment]
Also I've tried to downgrade to 0.96.4 and 0.94.2 (pip3 install homeassistant==0.94.2) and results are the same. Automations work for 1 minute, I can see them in logs, but after that nothing. I'm starting to think that maybe this new Xiaomi Hub/gw update is to blame. Could you please check in Mi Home app do you have any updates pending? My version is 1.4.1_170.0158. 

Also whats the maximum number of supported devices on xiaomi hub. I have 25?!

Thank you alot for your help, I will post debug logs in a few minutes...

**Comment by syssi** [comment]
Which version of the Gateway to you own? Are you sure your network (multicast) isn't the problem? I dont't know the exact limit anymore. It's around 30. Please provide some logs.

**Comment by llonca13** [comment]
I doubt its the network because it all worked great for many months. Also whats the chance that only the aqara square wouldn't work? 

And yes, I would love to give you logs but thats the problem. There are no logs.
After restart approx 2 mins automations work:
2019-10-04 20:47:39 DEBUG (Thread-5) [xiaomi_gateway] MCAST (report) << {'cmd': 'report', 'model': 'sensor_switch.aq3', 'sid': '158d0002adea2b', 'short_id': 18617, 'data': '{"status":"click"}'}
2019-10-04 20:47:39 DEBUG (MainThread) [homeassistant.components.xiaomi_aqara] PUSH >> <Entity Switch_158d0002adea2b: off>: {'status': 'click'}
2019-10-04 20:47:42 DEBUG (Thread-5) [xiaomi_gateway] MCAST (report) << {'cmd': 'report', 'model': 'sensor_switch.aq3', 'sid': '158d0002adea2b', 'short_id': 18617, 'data': '{"status":"double_click"}'}

But after that.. nothing.. no logs, no automations..

**Comment by syssi** [comment]
I fear the button is the problem. I don't own this hardware version to reproduce the issue.

**Comment by syssi** [comment]
The other devices are working flawlessly?

**Comment by llonca13** [comment]
Yes, for 1 year no problem at all. And I haven't change or add anything for 6 months minimum :/

Still thanks alot for your trouble

**Comment by syssi** [comment]
Did you update your gateway firmware in the meantime or did you block the internet access of the gateway?

**Comment by llonca13** [comment]
I never blocked internet access. I did update to the latest fw (1.4.1_170.0158) maybe 3 months ago, but I can assure you everything worked for some time(approx. 1 month) after that. I was pending of bugs after update.

**Comment by syssi** [comment]
This is strange. How much aq3 buttons do you own?

**Comment by llonca13** [comment]
3 aqara buttons :/

I know, Im going crazy and I cant believe nobody else has this problems (except 1-2 people on reddit). Guess this button version was not very popular (more expensive then non gyro version). I think I will throw them in the trash and order Aqara switches instead.

I also have deconz stick (in my rpi3 but its too far from my bedroom buttons), still Ill try if they work through deconz tomorrow.

**Comment by syssi** [comment]
All of them has this problem? Did you put the button next to the gateway for testing? Try to avoid zigbee repeater (just for testing: plug them off). There are zigbee repeaters around with broken routing tables f.e.

**Comment by syssi** [comment]
~Every wall powered zigbee device acts as repeater.

**Comment by llonca13** [comment]
I only have xiaomi zigbee devices and I think I read somewhere that they dont follow specification correctly, and that their in-wall devices dont work as routers.. so they all connect directly to hub.

Yes, all the buttons (WXKG12LM) has this weird problem.

I will try tomorrow again, its 12pm here.. thank you once again for all your free time wasted on me :/

**Comment by syssi** [comment]
No problem. Please post need findings here!

**Comment by llonca13** [comment]
I connected them through deconz and they work perfectly. Single, double, triple, quadruple?! clicks all work great. Please anybody else having the same problems as me - post here.

**Comment by llonca13** [comment]
Also through deconz I found out that one of my 2 buttons that weren't working was in fact non gyro square (WXKG11LM) version.
So for me both WXKG12LM and WXKG11LM stopped working for some mysterious reason :/

**Comment by KmanOz** [comment]
Hi I have only just acquired a Mi Hub Gateway and various sensors switches etc. I can confirm that the WXKG11LM does not work in HA but no problems in the Mi Home app.
